### PR TITLE
Feature/teams

### PR DESCRIPTION
**Pull request by lionellloh**
CRUD for teams 
RSpec 
Cucumber

### CODE REVIEW

File: app/controllers/teams_controller.rb [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    @team = Team.find(params[:id])
+    p @team
+  end
+
+  def new
+    @team = Team.new
+  end
+
+  def create
+    @team = Team.create(team_params)
```
Comment (ntiner1) [inline app/controllers/teams_controller.rb]: Make sure to seperate create into `new` and `save`. Should look like:
`@team = Team.new(team_params)`
`if @team.save`
`  redirect_to teams_path`
`else`
`  render 'new'`
`end`
This is incase the team isn't valid

File: features/step_definitions/team_steps.rb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+require File.expand_path(File.join(File.dirname(__FILE__), "..", "support", "selectors"))
+
+module WithinHelpers
+  def with_scope(locator)
+    locator ? within(*selector_for(locator)) { yield } : yield
+  end
+end
+World(WithinHelpers)
+
+When(/^I go create a team/) do
```
Comment (reedhampton) [inline features/step_definitions/team_steps.rb]: These step definitions are already defined but generically in the web_steps.rb file. That way we don't have to hardcode the button names etc.

We can add the page in paths.rb and simply use the web_steps versions for smaller code files

File: app/controllers/teams_controller.rb [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+    redirect_to teams_path
+  end
+
+  def update
+    @team = Team.find(params[:id])
+    @team.update(team_params)
```
Comment (ntiner1) [inline app/controllers/teams_controller.rb]: Similar to create.

File: app/controllers/teams_controller.rb [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+    redirect_to team_path (@team)
+  end
+
+  def destroy
+    @team = Team.find(params[:id])
+    @team.destroy
```
Comment (ntiner1) [inline app/controllers/teams_controller.rb]: Similar to create.

File: app/views/teams/edit.html.erb [hunk 5]
```diff
@@ -0,0 +1,34 @@
+<h3>Update Team Details</h3>
+<%= form_with model: @team do |form| %>
```
Comment (ntiner1) [inline app/views/teams/edit.html.erb]: If the form is the same in both edit and new, then move the form to `_form.html.erb`. Look at projects or skills form if u need an example.

File: app/views/teams/new.html.erb [hunk 6]
```diff
@@ -0,0 +1,33 @@
+<h3>Create a Team</h3>
+<%= form_with model: @team do |form| %>
```
Comment (ntiner1) [inline app/views/teams/new.html.erb]: Look at edit.html.erb for _form comment.

File: features/support/paths.rb [hunk 7]
```diff
@@ -43,6 +43,15 @@ def path_to(page_name)
     when /^the create skill\s?page$/
       '/skills/new'
 
+    when /^the create skill\s?page$/
```
Comment (ntiner1) [inline features/support/paths.rb]: Looks like this was accidentally repeated.

**Pull request closed without merging**